### ISSUE TITLE
Only allow clients to start booking with a provider if there are services to book

### ISSUE THREAD

**Issue by situatedbit** [loconomics/loconomics#137]
In #122 we decided that when a non-logged in user or a logged-in client searches for a provider but that provider does not have any services that are bookable by the public, or by the logged-in client specifically, they should still see the provider in the search result. 

However, currently the client is able to start the booking process with a provider who is not offering any services available to them. The _Book_ button is always available.

The _Book_ button should not be available to a client unless the provider has services that the client may book.

**Comment by situatedbit** [comment]
_Question_: if the user is anonymous (not logged in), should they see the book button? Probably, yes. We'd need a way to route them.

**Comment by joshdanielson** [comment]
@situatedbit yes. They create a login in the last step of the booking (after they're more invested in the booking).

**Comment by IagoSRL** [comment]
*Answer*: yes, if the professional has services of type 'anyone' and 'first-time-only'. The booking has embedded sign-up (last step), and link to log-in (with return to booking); if user choose to log-in, the booking updates the list of services available from the professional for the client, and any selection of services done at the booking before click log-in (if any) are double checked to be deselected or kept depending on the updated list.

*Edge case*: A professional has services for first-time-only clients but not for the client once gets logged.
